cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java

```diff
@@ -140,6 +147,46 @@ public void testBrokerDiesWhenMovePartitions() throws Exception {
     assertEquals(initialLeader0, zkUtils.getLeaderForPartition(TOPIC_1, PARTITION).get());
   }
 
+  @Test
+  public void testTimeoutLeaderActions() throws InterruptedException {
+    ZkUtils zkUtils = KafkaCruiseControlUnitTestUtils.zkUtils(zookeeper().getConnectionString());
+
+    Map<String, TopicDescription> topicDescriptions = createTopics();
+    // The proposal tries to move the leader. We fake the replica list to be unchanged so there is no replica
+    // movement, but only leader movement.
+    ExecutionProposal proposal =
+        new ExecutionProposal(TP1, 0, 1, Arrays.asList(0, 1), Arrays.asList(0, 1));
+    System.out.println(proposal);
+
+    KafkaCruiseControlConfig configs = new KafkaCruiseControlConfig(getExecutorProperties());
+    Time time = new MockTime();
+    MetadataClient mockMetadataClient = EasyMock.createMock(MetadataClient.class);
+    Node node0 = new Node(0, "host0", 100);
+    Node node1 = new Node(1, "host1", 100);
+    Node[] replicas = new Node[2];
+    replicas[0] = node0;
+    replicas[1] = node1;
+    PartitionInfo partitionInfo = new PartitionInfo(TP1.topic(), TP1.partition(), node1, replicas, replicas);
+    Cluster cluster = new Cluster("id", Arrays.asList(node0, node1), Collections.singleton(partitionInfo),
+                                  Collections.emptySet(), Collections.emptySet());
+    MetadataClient.ClusterAndGeneration clusterAndGeneration = new MetadataClient.ClusterAndGeneration(cluster, 0);
+
+    EasyMock.expect(mockMetadataClient.refreshMetadata()).andReturn(clusterAndGeneration).anyTimes();
+    EasyMock.expect(mockMetadataClient.cluster()).andReturn(clusterAndGeneration.cluster()).anyTimes();
+    EasyMock.replay(mockMetadataClient);
+
+    Collection<ExecutionProposal> proposalsToExecute = Collections.singletonList(proposal);
+    Executor executor = new Executor(configs, time, new MetricRegistry(), mockMetadataClient);
+    executor.executeProposals(proposalsToExecute,
+                              Collections.emptySet(),
+                              EasyMock.mock(LoadMonitor.class));
+    while (executor.state().state() != ExecutorState.State.LEADER_MOVEMENT_TASK_IN_PROGRESS) {
+      Thread.sleep(10);
+    }
+    time.sleep(180001);
```

Comment:
Will this add 3 minutes to our unit test time? If so, is there a way to mock this?